transpiler/src/ast/core/intermediate-ast/builders/controllers/restController/RESTControllerNodeBuilder.ts

```diff
@@ -0,0 +1,59 @@
+import { RESTControllerNode } from '../../../nodes/controllers/restController/RESTControllerNode.js';
+import { IntermediateASTTree } from '../../../IntermediateASTTree.js';
+import { TNodeMetadata } from '../../../nodes/IntermediateASTNode.js';
+import { IBuilder } from '../../IBuilder.js';
+import { RESTControllerIdentifierNode } from '../../../nodes/controllers/restController/RESTControllerIdentifierNode.js';
+import { RESTControllerExecuteNode } from './../../../nodes/controllers/restController/RESTControllerExecuteNode.js';
+import { ParameterListNode } from '../../../nodes/ParameterList/ParameterListNode.js';
+import { RESTMethodNode } from './../../../nodes/controllers/restController/RESTMethodNode.js';
+
+export class RESTControllerNodeBuilder implements IBuilder<RESTControllerNode> {
+  private restControllerNode: RESTControllerNode;
+  private restControllerIdentifierNode: RESTControllerIdentifierNode;
+  private parameters: ParameterListNode;
+  private restMethod: RESTMethodNode;
+  private controllerExecuteNode: RESTControllerExecuteNode;
+  private intermediateASTTree: IntermediateASTTree;
+
+  constructor(intermediateASTTree: IntermediateASTTree, metadata?: TNodeMetadata) {
+    this.intermediateASTTree = intermediateASTTree;
+    this.restControllerNode = new RESTControllerNode(metadata);
+  }
+
+  public withIdentifier(identifierNode: RESTControllerIdentifierNode): RESTControllerNodeBuilder {
+    this.restControllerIdentifierNode = identifierNode;
+    const controllerName = identifierNode.getIdentifierName();
+    this.restControllerNode.setClassName(controllerName);
+    return this;
+  }
+
+  public withParameterList(parametersList: ParameterListNode): RESTControllerNodeBuilder {
+    this.parameters = parametersList;
+    return this;
+  }
+
+  withRESTMethod(restMethod: RESTMethodNode): RESTControllerNodeBuilder {
```

Comment:
minor but if you can add public for stability(maybe in the next PR)

Reply:
on it!